### ISSUE TITLE
✨ Enhancement: Special sidebar behaviour for pages for pages without sidebar item

### ISSUE THREAD

**Issue by benjagm** [json-schema-org/website#1011]
### Is your feature request related to a problem? Please describe

We have some pages without sidebar item that needs to be shown, however when we show those pages it results in a sidebar with all sections collapsed. For example, the page https://json-schema.org/md-style-guide produces:

<img width="1245" alt="Screenshot 2024-10-09 at 06 37 23" src="https://github.com/user-attachments/assets/44879fcc-3559-42af-bcff-2cb62622bab5">

This is not ideal and we would like to find a better way to interact with the sidebar for those pages.

### Describe the solution you'd like

Ideally, we'd like to define a new property in the header metadata in the page markdown to specify the parent page to be highlighted in the sidebar.

For example, if we define this new property as parentNav adding the relative path for a page without sidebar item like https://json-schema.org/md-style-guide:
````
---
title: Markdown Style Guide
parentNav: /overview/what-is-jsonschema
---
````

The expected result is to highlight /overview/what-is-jsonschema in the sidebar when accessing that page. See image below:
<img width="289" alt="Screenshot 2024-10-09 at 06 42 25" src="https://github.com/user-attachments/assets/f0c7ce6b-d7ad-4431-a52b-b26bbdebc284">

### Describe alternatives you've considered

I think this will require:

- Define the new property in the markdown metadata.
- Define a mechanism to read that property in the typescript code reading the markdown.
- Modify DocLink function in the sidebar logic to support highlighting the referred item. 

### Are you working on this?

No

**Comment by pavanydg** [comment]
@benjagm I would like to work on this.

**Comment by benjagm** [comment]
> @benjagm I would like to work on this.

Thanks a lot for your support!! Please go ahead!

**Comment by DhairyaMajmudar** [comment]
@pavanydg any progress on this ? Or any help needed ?

**Comment by pavanydg** [comment]
@DhairyaMajmudar sorry for the delay. I was working on another issue in tour site. Will update about the progress within 2 days.

**Comment by DhairyaMajmudar** [comment]
> @DhairyaMajmudar sorry for the delay. I was working on another issue in tour site. Will update about the progress within 2 days.

No worries, pls. take your time

**Comment by pavanydg** [comment]
@DhairyaMajmudar so when we open this page https://json-schema.org/md-style-guide we should highlight overview/what-is-json-shema in sidebar right? I should implement only this much right?

**Comment by DhairyaMajmudar** [comment]
> @DhairyaMajmudar so when we open this page https://json-schema.org/md-style-guide we should highlight overview/what-is-json-shema in sidebar right? I should implement only this much right?

yep, also pls. check for such more pages whose direct link is not given in the sidebar :)

**Comment by DhairyaMajmudar** [comment]
Thank you @techmannih for showing your interest but it seems you already have a good number of issues assigned here and in conference repo. please resolve them first taking new one.

Also @pavanydg what are the updates on this, you need any help ?

**Comment by pavanydg** [comment]
@DhairyaMajmudar @benjagm  i need help on this one. Like i have to extract the data from .md files to check if it has parentNav or not to check whether this is a sidebaritem or not. (this can be done using getStaticMarkdownProps)
I can do this but won't it be unecessary as almost all pages won't have parentNav.
So please give me some suggestions on how i should do it.

**Comment by Karan-Palan** [comment]
Hello @DhairyaMajmudar and @benjagm , this issue seems to have no updates. I would like to take this up!

**Comment by DhairyaMajmudar** [comment]
@pavanydg if you're not working on this shall I assign this issue to Karan ?

**Comment by pavanydg** [comment]
@Karan-Palan I am not working on this right now. You can take it up.

**Comment by Karan-Palan** [comment]
> @Karan-Palan I am not working on this right now. You can take it up.

Sure I'll wait for an assignment from @DhairyaMajmudar

**Comment by DhairyaMajmudar** [comment]
@Karan-Palan still interested in taking this up?

**Comment by Karan-Palan** [comment]
Yeah! Just done with exams, have ample of time

**Comment by Utkarsh-123github** [comment]
@DhairyaMajmudar  @benjagm  , this issue has been resolved as we can observe the requested functionality on our website. So, I think we can close this issue.

**Comment by Utkarsh-123github** [comment]
Ah, I see! I believe I may have misinterpreted what I observed earlier—apologies for that :)

**Comment by benjagm** [comment]
@Utkarsh-123github  can you please work on this issue?

**Comment by Utkarsh-123github** [comment]
@benjagm I'd love to work on this 🚀

**Comment by benjagm** [comment]
Hi @Utkarsh-123github , as discussed let's do the following:

- In this case remove the sidebar.
- For the release notes subsections mark the release notes item in the sidebar: https://json-schema.org/specification/release-notes
